python/gtsam/examples/EqF.py

```diff
@@ -0,0 +1,887 @@
+    cal_exist = False
+    n_cal = 0
+    for i in range(6):
+        if {'cq_x_' + str(i), 'cq_y_' + str(i), 'cq_z_' + str(i), 'cq_w_' + str(i)}.issubset(df.columns):
+            cal_exist = True
+            n_cal = i+1
+
+    # Check for existence of direction measurements
+    n_meas = 0
+    for i in range(11):
+        if {'y_x_' + str(i), 'y_y_' + str(i), 'y_z_' + str(i)}.issubset(df.columns):
+            n_meas = i + 1
+
+    for index, row in df.iterrows():
+
+        # Load timestamps and record dt
+        t = float(row['t'])
+        dt = t - last_timestamp
+
+        # Skip data_list if dt is smaller than a micro second
+        if dt < 1e-6:
+            continue
+
+        last_timestamp = t
+
+        # Load groundtruth values
+        quat = np.array([float(row['q_x']), float(row['q_y']), float(row['q_z']), float(row['q_w'])])
+
+        R = Rot3(gtsam.Rot3.Quaternion(float(row['q_w']), float(row['q_x']), float(row['q_y']), float(row['q_z'])).matrix())
+
+        # Load Gyro biases
+        if bias_exist:
+            b = np.array([float(row['b_x']), float(row['b_y']), float(row['b_z'])]).reshape(3,)
+        else:
+            b = np.zeros(3)
+
+        # Load GNSS calibration
+        S = []
+        if cal_exist:
+            for i in range(n_cal):
+                cal = np.array([float(row['cq_x_' + str(i)]), float(row['cq_y_' + str(i)]), float(row['cq_z_' + str(i)]), float(row['cq_w_' + str(i)])])
+                S.append(Rot3(gtsam.Rot3.Quaternion(float(row['cq_w_' + str(i)]), float(row['cq_x_' + str(i)]), float(row['cq_y_' + str(i)]), float(row['cq_z_' + str(i)])).matrix()))
+
+
+        # Load Gyro inputs
+        w = np.array([float(row['w_x']), float(row['w_y']), float(row['w_z'])]).reshape(3,)
+        std_w = np.array([float(row['std_w_x']), float(row['std_w_y']), float(row['std_w_z'])]).reshape(3,)
+        std_b = np.array([float(row['std_b_x']), float(row['std_b_y']), float(row['std_b_z'])]).reshape(3,)
+        Sigma_wb = blockDiag(np.eye(3) * (std_w ** 2), np.eye(3) * (std_b ** 2))
+
+        # Load measurements
+        meas = []
+        for i in range(n_meas):
+            y = np.array([float(row['y_x_' + str(i)]), float(row['y_y_' + str(i)]), float(row['y_z_' + str(i)])]).reshape(3,)
+            d = np.array([float(row['d_x_' + str(i)]), float(row['d_y_' + str(i)]), float(row['d_z_' + str(i)])]).reshape(3,)
+            std_y = np.array([float(row['std_y_x_' + str(i)]), float(row['std_y_y_' + str(i)]), float(row['std_y_z_' + str(i)])]).reshape(3,)
+            if i < n_cal:
+                meas.append(Measurement(y, d, np.eye(3) * (std_y ** 2), i))
+            else:
+                meas.append(Measurement(y, d, np.eye(3) * (std_y ** 2), -1))
+
+        # Append to data_list
+        d = Data(State(R, b, S), n_cal, Input(w, Sigma_wb), meas, n_meas, t, dt)
+        
+        data_list.append(d)
+
+    return data_list
+
+
+def sim(filter_args, data):
```

Comment:
I propose you leave this code in the notebook (and the corresponding plotting imports)